Template Color updates

davidortinau opened the pull request:
I propose not using Light/Dark for color names we provide, and instead using Primary, Secondary, Tertiary. Let's leave Light/Dark to AppThemeBinding and if the dev wants to add semantic color names.

To replace the grays, I propose using names that indicate hue. This also matches the Fluent color guideline.

davidortinau committed:
Color updates

I propose not using Light/Dark for color names we provide, and instead using Primary, Secondary, Tertiary. Let's leave Light/Dark to AppThemeBinding and if the dev wants to add semantic color names.

To replace the grays, I propose using names that indicate hue. This also matches the Fluent color guideline.

Eilon commented:
The colors in practice seem quite different from what they were before. Can you show screenshot before/after for this? I'm wondering whether the contrast is still sufficient for accessibility needs.

Example: Where it's consumed, `DarkGray` was replaced with `Gray950`, but the actual color changed from `#505050` to `#141414`, which is significantly darker.

Before:
<img width="237" alt="lighter dark color" src="https://user-images.githubusercontent.com/202643/167919815-eb7aec14-c4ab-4649-9b6d-a37c2662ed61.png">

After:
<img width="214" alt="darker dark color" src="https://user-images.githubusercontent.com/202643/167919735-db642e5e-a27d-43eb-a294-c201bc123cd8.png">

davidortinau commented:
@Eilon will do.

rachelkang committed:
Merge branch 'main' into colors

Redth commented on src/Templates/src/templates/maui-mobile/Resources/Styles/Styles.xaml:
```diff
@@ -345,15 +345,15 @@
         <Setter Property="Shell.BackgroundColor" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource Black}}" />
-        <Setter Property="Shell.ForegroundColor" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource White}}" />
-        <Setter Property="Shell.TitleColor" Value="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource White}}" />
+        <Setter Property="Shell.ForegroundColor" Value="{OnPlatform WinUI={StaticResource Primary}, Default={StaticResource White}}"></Setter>
```
```suggestion
        <Setter Property="Shell.ForegroundColor" Value="{OnPlatform WinUI={StaticResource Primary}, Default={StaticResource White}}" />
```

Just a nit on introducing the closing tag

rachelkang committed:
Tweak Shell and TimePicker styles

Eilon commented:
Nifty, thank you for checking this!

davidortinau committed:
Fluent color revisions

added label disabled, changed bg of controls to transparent where page should bleed through

davidortinau committed:
Merge branch 'colors' of github.com:dotnet/maui into colors

rachelkang committed:
Final tweaks
